src/compiler.ts

```diff
@@ -6597,12 +6636,14 @@ export class Compiler extends DiagnosticEmitter {
               break;
             }
             case TypeKind.USIZE: {
-              if (this.currentType.is(TypeFlags.REFERENCE)) {
-                this.error(
-                  DiagnosticCode.Operation_not_supported,
-                  expression.range
-                );
-                return module.createUnreachable();
+              // check operator overload
+              let classReference = this.currentType.classReference;
+              if (classReference) {
+                let overload = classReference.lookupOverload(OperatorKind.MINUS);
+                if (overload) {
+                  expr = this.compileUnaryOverload(overload, expression.operand, expression);
+                  break;
+                }
```

Comment:
Hmm, this should probably still result in an "Operation not supported" error if there is no overload defined. Wdyt? I think this affects pretty much all of the overloads.

Reply:
Yes, makes sense. Fixed!

Reply:
Also make sure `usize` is actually reference and skipping error if it just usize type with arithmetics.